src/main/java/org/openstreetmap/atlas/checks/validation/intersections/BigNodeBadDataCheck.java

```diff
@@ -77,29 +78,42 @@ public boolean validCheckForObject(final AtlasObject object)
         {
             return Optional.empty();
         }
-        final int allPathsCount = bigNode.allPaths().size();
+
         final int junctionEdgesCount = bigNode.junctionEdges().size();
-        final Set<RestrictedPath> turnRestrictions = bigNode.turnRestrictions();
-        final int turnRestrictionCount = turnRestrictions.size();
-        // Get list of restricted path OSM ids
-        final List<Long> restrictedPathOsmIds = turnRestrictions.stream()
-                .flatMap(restrictedPath -> Iterables.asList(restrictedPath.getRoute()).stream()
-                        .map(Edge::getOsmIdentifier))
-                .collect(Collectors.toList());
-
-        if (allPathsCount > maxNumberPathsThreshold
-                || junctionEdgesCount > maxNumberJunctionEdgesThreshold)
+        if (junctionEdgesCount > maxNumberJunctionEdgesThreshold)
         {
             final CheckFlag flag = createFlag(bigNode,
-                    this.getLocalizedInstruction(0, allPathsCount, junctionEdgesCount,
-                            turnRestrictionCount, restrictedPathOsmIds.toString()));
+                    this.getLocalizedInstruction(0, junctionEdgesCount));
 
             bigNode.edges().forEach(flag::addObject);
             // Add map pin at each BigNode Node
             bigNode.nodes().forEach(b -> flag.addPoint(b.getLocation()));
 
             return Optional.of(flag);
         }
+        else
```

Comment:
@Bentleysb Great improvement on the run time. I only have one doubt. So, here are you trying to get bigNode.allPaths() only if no other flag conditions are met so that you can save run time!!

Reply:
That is the idea. Although currently the number of junction edges is the only other flag condition.

Reply:
So we cannot avoid running into the case where it takes lot of time going through all the big node paths but we can at least save time on checking conditions ahead like junction edges. Cool. Thanks.